docs/emulate.rst

```diff
@@ -0,0 +1,62 @@
+.. _emulate:
+
+
+Prepare QEMU
+------------
+
+You will need to prepare QEMU STM32 emulator from:
+`QEMU STM32 v0.1.3 <https://github.com/beckus/qemu_stm32/releases>`_
+
+After download, compile the qemu_stm32 by::
+
+    $ ./configure --disable-werror --enable-debug \
+            --target-list="arm-softmmu" \
+            --extra-cflags=-DDEBUG_CLKTREE \
+            --extra-cflags=-DDEBUG_STM32_RCC \
+            --extra-cflags=-DDEBUG_STM32_UART \
+            --extra-cflags=-DSTM32_UART_NO_BAUD_DELAY \
+            --extra-cflags=-DSTM32_UART_ENABLE_OVERRUN --python=python2
+    $ make -j8
+
+
+Make sure you have export the `arm-softmmu` to `$PATH`::
+
+    $ export PATH=$PATH:~/qemu_stm32-stm32_v0.1.3/arm-softmmu
+
+
+Run Piko/RT on QEMU STM32-p103
+------------------------------
+
+You will need to run these command::
+
+    $ make TARGET=stm32p103
+    $ make TARGET=stm32p103 run
```

Comment:
Exactly, current implementation lacks of the necessary to validate QEMU's capability (we need the patched and non-official one!). It would be nice if we can modularize QEMU scripts and share with test suite as Openmoko did: http://svn.openmoko.org/trunk/src/host/qemu-neo1973/openmoko/ (It is 10 years ago though.)